docs/source/production.rst

```diff
@@ -85,3 +85,32 @@ such as:
   lines). The default is `origin/master`.
 * Configuring `diff-quality` to return an error code if the quality is too low
 * Troubleshooting
+
+Using `pre-commit`_
+^^^^^^^^^^^^^^^^^^^
+
+SQLFluff comes with two `pre-commit`_ hooks:
+
+* sqlfluff-lint: returns linting errors.
+* sqlfluff-fix: attempts to fix rule violations.
+
+Running with `pre-commit`_, you get the benefit of
+only linting/fixing the files that changed.
```

Comment:
Great feature, however I think this needs to be fleshed out a little. My assumption is that a lot of sql users (and therefore sqlfluff users) won't know what a `pre-commit` hook is and when to use one. Either we should flesh out a little more detail here on when you might  want to use this feature or we should link to some external literature on that topic, just so people can make informed decisions about when and how to use these.